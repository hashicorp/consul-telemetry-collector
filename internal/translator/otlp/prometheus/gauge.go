package prometheus

import _go "github.com/prometheus/client_model/go"

func NewGauge(family _go.MetricFamily) {
	_ = _go.MetricFamily{}
}
