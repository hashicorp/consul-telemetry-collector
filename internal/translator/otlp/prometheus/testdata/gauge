[
  {
    "name": "cluster.upstream_cx_active",
    "type": 1,
    "metric": [
      {
        "label": [
          {
            "name": "envoy.cluster_name",
            "value": "service_envoyproxy_io"
          }
        ],
        "gauge": {
          "value": 9
        },
        "timestamp_ms": 1682099630190
      }
    ]
  },
  {
    "name": "cluster.upstream_cx_active",
    "type": 1,
    "metric": [
      {
        "label": [
          {
            "name": "envoy.cluster_name",
            "value": "metrics_cluster"
          }
        ],
        "gauge": {
          "value": 1
        },
        "timestamp_ms": 1682099630190
      }
    ]
   },
   {
    "name": "server.stats_recent_lookups",
    "type": 1,
    "metric": [
      {
        "gauge": {
          "value": 2541
        },
        "timestamp_ms": 1682099630190
      }
    ]
  }
]