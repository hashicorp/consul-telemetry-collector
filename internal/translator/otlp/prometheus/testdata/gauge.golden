{
  "resourceMetrics": [
    {
      "resource": {},
      "scopeMetrics": [
        {
          "scope": {},
          "metrics": [
            {
              "name": "envoy_cluster_upstream_cx_active",
              "gauge": {
                "dataPoints": [
                  {
                    "attributes": [
                      {
                        "key": "envoy_cluster_name",
                        "value": {
                          "stringValue": "metrics_cluster"
                        }
                      }
                    ],
                    "timeUnixNano": "1682099630856000000",
                    "asDouble": 1
                  },
                  {
                    "attributes": [
                      {
                        "key": "envoy_cluster_name",
                        "value": {
                          "stringValue": "service_envoyproxy_io"
                        }
                      }
                    ],
                    "timeUnixNano": "1682099630856000000",
                    "asDouble": 9
                  }
                ]
              }
            },
            {
              "name": "envoy_server_stats_recent_lookups",
              "gauge": {
                "dataPoints": [
                  {
                    "attributes": [],
                    "timeUnixNano": "1682099630856000000",
                    "asDouble": 2541
                  }
                ]
              }
            }
          ]
        }
      ]
    }
  ]
}
